### PR TITLE
Add tests for including unprocessed samples in get_current_speech

### PR DESCRIPTION
**Pull request by WenqingZong**
As title

General idea: if `get_current_speech` includes unprocessed samples, then the last speech start timestamp + duration of current_speech should be equal to the total duration we send to vad.

### CODE REVIEW

File: tests/consistency.rs [hunk 1]
```diff
@@ -39,6 +39,109 @@ fn compare_birds() {
...
+        // Update last_speech_start_ms
+        for transition in transitions {
+            if let VadTransition::SpeechStart { timestamp_ms } = transition {
+                last_speech_start_ms = timestamp_ms;
+            }
+        }
+
+        // The actual test logic.
+        if vad.is_speaking() {
+            let current_speech = vad.get_current_speech();
```
Comment (xd009642) [inline tests/consistency.rs:131]: Do we know that this branch is ever exercised. Because if it isn't we'll be adding a test that does nothing which isn't too useful.

Indeed birds has no parts where it shows as `is_speaking` so that test does nothing :eyes:

File: tests/consistency.rs [hunk 2]
```diff
@@ -39,6 +39,100 @@ fn compare_birds() {
...
+        let transitions = vad.process(&samples[start..end]).unwrap();
+        // Update last_speech_start_ms
+        for transition in transitions {
+            if let VadTransition::SpeechStart { timestamp_ms } = transition {
+                last_speech_start_ms = timestamp_ms;
+            }
+        }
+
+        // The actual test logic.
+        if vad.is_speaking() {
```
Comment (xd009642) [inline tests/consistency.rs:121]: So follow on from the last review comment. There's no point testing with chunk size of 30 because there's no remainder then so nothing will be in unprocessed. Maybe it would be good to try a wackier size like 47 (prime number and over the window size). 

Also, to ensure we don't end up with useless tests that never hit the branch you should probably add a bool to the test that goes from false to true when this condition is entered and at the end `assert!(conditions_tested, "test was useless");` 

Adding a `#[doc(hidden)]` for the current unprocessed duration as well and asserting that goes >0 during the test would also be a nice addition that makes it even more robust.

**Comment by WenqingZong** [comment]
```text
Wenqing Zong
  12:07 AM
Got test failures, unprocessed samples is always 0


Daniel 泥妮
  12:08 AM
is that when you still have the 30ms chunk size?


Wenqing Zong
  [12:08 AM](https://ollydevelopment.slack.com/archives/D05QDK7HVAT/p1733760503551699)
no, tried 47, and still failure


Daniel 泥妮
  12:10 AM
hmmm but we do leave the remainder behind don't we? Because if we do different chunk sizes we now get the same output :thethinker:


Wenqing Zong
  [12:10 AM](https://ollydevelopment.slack.com/archives/D05QDK7HVAT/p1733760627349559)
image.png
 
image.png

Wenqing Zong
  12:15 AM
        for i in 0..num_chunks {
            let sample_range = if i < num_chunks - 1 {
                (self.processed_samples - self.deleted_samples)
                    ..(self.processed_samples + vad_segment_length - self.deleted_samples)
            } else {
                (self.processed_samples - self.deleted_samples)..self.session_audio.len()
            };
        // The rest of process function...
        }
Here, process  function process everything in session_audio, so after this function, unprocessed is always 0.
[12:18](https://ollydevelopment.slack.com/archives/D05QDK7HVAT/p1733761080772779)
the remainder only exists before a call to process  finishes, if I understand it correctly now.


Daniel 泥妮
  12:20 AM
oh hmm, well in that case making it keep the remainder and not process it would make the vad deterministic - if we're in an is_speaking at the end of audio and include the remainder it'll always be <30ms so no harm including it
[12:21](https://ollydevelopment.slack.com/archives/D05QDK7HVAT/p1733761274299329)
I wonder if that would help in the edge cases we occasionally see - just through making things more deterministic.
12:21
But anyway until the library actually does that the test can't be added and pass so you can shelve it and go back to looking at normalisation if you want


Wenqing Zong
  12:22 AM
ok, I'll close that pr then.
```

Close as the above reason.
